### PR TITLE
Add docker private registry support

### PR DESCRIPTION
**Pull request by secat**
This PR addresses #43 and add support for a Docker Private Registry deployed on premise.

### CODE REVIEW

**Comment by secat** [comment]
I am currently working on fixing the tests.

**Comment by stevesloka** [comment]
@secat I'll check this out soon, just had a baby last week so a bit behind. Thanks!

**Comment by secat** [comment]
@stevesloka No problem, I was exactly in the same situation last year! I fully understand your situation!

**Comment by stevesloka** [comment]
Hey @secat would you mind rebasing off of master? There are some conflicts now, then I'll do a quick test and get this out. Also I can add support in minikube's addon UI.

**Comment by secat** [comment]
Ok, I will work on it today.

**Comment by secat** [comment]
@stevesloka Friendly reminder: this PR is ready for review and has been rebase with the latest changes.

**Comment by stevesloka** [comment]
Why did the makefile change so much? When I try and just build I get the following: 

```
 steve@phippy  ~/godev/src/github.com/upmc-enterprises/registry-creds   add-docker-private-registry-support  make build
Building: bin/registry-creds
GOARCH=amd64 CGO_ENABLED=0 go install
go install os/user: open /usr/local/go/pkg/linux_amd64/os/user.a: permission denied
go install net: open /usr/local/go/pkg/linux_amd64/net.a: permission denied
Makefile:37: recipe for target 'bin/registry-creds' failed
make: *** [bin/registry-creds] Error 1
```

**Comment by stevesloka** [comment]
I get the same on my mac:

```
~/g/s/g/u/registry-creds ❯❯❯ make build                                                                       add-docker-private-registry-support ◼
Building: bin/registry-creds
GOARCH=amd64 CGO_ENABLED=0 go install
go install os/user: open /usr/local/go/pkg/darwin_amd64/os/user.a: permission denied
go install net: open /usr/local/go/pkg/darwin_amd64/net.a: permission denied
make: *** [bin/registry-creds] Error 1
```

File: Makefile [hunk 1]
```diff
@@ -2,24 +2,66 @@
...
+BIN = registry-creds
+ARCH = amd64
+
+# go option
+GO      ?= go
+LDFLAGS := -w
+GOFLAGS := -a -installsuffix cgo
+
+# docker build arguments for internal proxy
+ifneq ($(http_proxy),)
```
Comment (stevesloka) [inline Makefile]: Is this to push to a local registry? I don't see why that's needed.

File: Makefile [hunk 2]
```diff
@@ -2,24 +2,66 @@
...
+.PHONY: all
 all: container
 
-build: main.go
-	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -a -installsuffix cgo -o registry-creds --ldflags '-w' ./main.go
+.PHONY: build
+build: build-dirs bin/$(BIN) bin/linux_$(ARCH)/$(BIN)
+
+# local developement binary (auto detect developer OS)
+bin/$(BIN): main.go
```
Comment (stevesloka) [inline Makefile]: Why `go install` each time? If I'm just building, I don't see the need to put into path.

File: README.md [hunk 3]
```diff
@@ -9,89 +10,119 @@ Allow for Registry credentials to be refreshed inside your Kubernetes cluster vi
...
 ## Developing Locally
 
 If you want to hack on this project:
 
 1. Clone the repo
-2. Build: `make binary`
+2. Build: `make build`
 3. Test: `make test`
-4. Run on your machine: ` go run ./main.go --kubecfg-file=<pathToKubecfgFile> --use-kubernetes-cluster-service=false
+4. Run on your machine: `./bin/registry-creds --kubecfg-file=<pathToKubecfgFile>` or `go run ./main.go --kubecfg-file=<pathToKubecfgFile>`
```
Comment (stevesloka) [inline README.md]: I don't like the idea of having to build it first, then run. I like the workflow to just `go run`, see the output, stop it, then go run again. Do you have thoughts as to why this is better? I ends up being two steps each time you want to build.

File: main.go [hunk 4]
```diff
@@ -43,21 +45,30 @@ import (
 )
 
 const (
+	dockerPrivateRegistryPasswordKey = "DOCKER_PRIVATE_REGISTRY_PASSWORD"
```
Comment (stevesloka) [inline main.go]: Can we combine with the other const?

File: main.go [hunk 5]
```diff
@@ -289,7 +355,7 @@ func main() {
 	validateParams()
 
 	log.Print("Using AWS Account: ", awsAccountID)
-	log.Printf("Using AWS Region: %s", *argAWSRegion)
+	log.Print("Using AWS Region: ", *argAWSRegion)
```
Comment (stevesloka) [inline main.go]: Why take out the printf?

File: Makefile (hunk 1 above)
Comment (secat) [inline Makefile]: It is to build the `dockerfile`: it will not build if working behind a corporate proxy.

File: Makefile (hunk 2 above)
Comment (secat) [inline Makefile]: This comes from [Stackoverflow - What does go build build](http://stackoverflow.com/questions/30612611/what-does-go-build-build):

> So basically you can use `go build` as a check that the packages can be built (along with their dependencies) while `go install` also (permanently) installs the results in the proper folders of your `$GOPATH`.

For developing, and iterating it is faster. However, I can revert that change.

File: main.go (hunk 4 above)
Comment (secat) [inline main.go]: Good idea, I will make the change.

File: main.go (hunk 5 above)
Comment (secat) [inline main.go]: In order to be consistent with other `Print`, I can revert this change.

File: README.md (hunk 3 above)
Comment (secat) [inline README.md]: Ok, I will rework my PR and remove these changes. The `Makefile` should look more like the original.

File: main.go (hunk 5 above)
Comment (stevesloka) [inline main.go]: Oh I see your point, you can keep it, makes sense!

**Comment by secat** [comment]
@stevesloka I addressed all previous comments.

I reverted almost all my changes in the Makefile. I am working on a windows platform and I did not realize that this project was favouring `go run` in order to run the binary outside of a docker container.

You should be able to build on your machine again. Sorry for that.

**Comment by stevesloka** [comment]
Hey @secat I just merged! Thanks so much for the PR! This is really helpful!

**Pull request merged**